### PR TITLE
Fixed benchmarks for the `0.25.3`

### PR DESCRIPTION
**Pull request by xgreenx**
Closes https://github.com/FuelLabs/fuel-core/issues/1807

The PR updates benchmarks to work properly with new VM rules and fixes the issue with non-optimal `InMemoryTransaction` behavior for the `size_of_value` method.

Because of the new VM rules: the memory that we are accessing should be initialized. The `cfe` opcode in the benchmarks expands the stack to allocate the memory that will later be used by the instruction that we are benchmarking.

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: benches/benches/vm_set/flow.rs [hunk 1]
```diff
@@ -70,7 +70,7 @@ pub fn run(c: &mut Criterion) {
             format!("{i}"),
             VmBench::contract(rng, op::retd(RegId::ONE, 0x10))
                 .unwrap()
-                .with_post_call(vec![op::movi(0x10, *i)])
+                .with_post_call(vec![op::movi(0x10, *i), op::cfe(0x10)])
```
Comment (bvrooman) [inline benches/benches/vm_set/flow.rs:73]: I'm not sure I understand the significance of adding `cfe` here or why it uses these specific registers. Can you elaborate on why we need this change? Is this just to make sure our benchmarks account for the worst case scenario?

**Review by bvrooman — APPROVED** [review]
Just a small optional correct and question

File: benches/benches/vm_initialization.rs [hunk 2]
```diff
@@ -85,6 +87,9 @@ pub fn vm_initialization(c: &mut Criterion) {
         if size as u64 > consensus_params.script_params().max_script_data_length() {
             break
         }
+        if 2 * size as u64 > consensus_params.tx_params().max_size() {
```
Comment (MitchTurner) [inline benches/benches/vm_initialization.rs:90]: Can we put this into a helper with the name that explains why this should `break`? Same with the above `break`.

```
if `something_size_is_foo_bar_baz(size, consensus_params)` {
    break
}
```

It's not clear to me what the significance of `size` (size of what?) being half as big as the max tx size.

**Review by bvrooman — APPROVED** [review]

File: benches/benches/vm_initialization.rs [hunk 3]
```diff
@@ -99,8 +104,36 @@ pub fn vm_initialization(c: &mut Criterion) {
                     Interpreter::<_, Script, NotSupportedEcal>::with_memory_storage(),
                 );
                 let ready_tx = tx.clone().test_into_ready();
-                black_box(vm.init_script(ready_tx))
-                    .expect("Should be able to execute transaction");
+
+                // Initialize the VM and require the allocation of the whole memory
+                // to charge for the worst possible case.
+                black_box({
```
Comment (MitchTurner) [inline benches/benches/vm_initialization.rs:110]: I could say a the same for a lot of this. Can we move everthing in these curlies into some helper?

File: crates/storage/src/transactional.rs [hunk 4]
```diff
@@ -249,6 +283,31 @@ where
             self.storage.get(key, column)
         }
     }
+
+    fn read(
```
Comment (MitchTurner) [inline crates/storage/src/transactional.rs:287]: I'm confused. Did we not have implementations for these methods before on `InMemoryTransaction`?

Comment (xgreenx) [reply crates/storage/src/transactional.rs:287]: We had a default implementation that was calling a `get` method which is less optimal in the case of RocksDB.

image

File: benches/benches/vm_initialization.rs [hunk 5]
```diff
@@ -75,36 +79,71 @@ fn transaction<R: Rng>(
...
+                let vm = black_box(
                     Interpreter::<_, Script, NotSupportedEcal>::with_memory_storage(),
                 );
-                let ready_tx = tx.clone().test_into_ready();
-                black_box(vm.init_script(ready_tx))
-                    .expect("Should be able to execute transaction");
+
+                // Initialize the VM and require the allocation of the whole memory
+                // to charge for the worst possible case.
+                black_box(initialize_vm(black_box(tx.clone()), vm));
```
Comment (MitchTurner) [inline benches/benches/vm_initialization.rs:110]: This is even more confusing from before. I'm not sure where the extra `black_box` came from either. 

Can we just have a method that says what we want, and then hide the implementation behind it? We probably can get rid of the comment in that case too.

`unoptimized_and_fully_saturated_vm` or something like that. Then have the internal code create the `vm` and wrap the `vm` and the `tx` in `black_box`?

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**